common/utils/include/claragenomics/utils/cudautils.hpp

```diff
@@ -89,6 +89,33 @@ __host__ __device__ __forceinline__
     return (value + boundary) & ~(boundary - 1);
 }
 
+template <typename Type>
+Type get_value_from_device(const Type* d_ptr, cudaStream_t stream = 0)
+{
+    Type val;
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(&val, d_ptr, sizeof(Type), cudaMemcpyDeviceToHost, stream));
+    CGA_CU_CHECK_ERR(cudaStreamSynchronize(stream));
+    return val;
+}
+
+template <typename Type>
+void set_device_value_async(Type* dst, const Type* src, cudaStream_t stream)
+{
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(dst, src, sizeof(Type), cudaMemcpyDefault, stream));
+}
+
+template <typename Type>
+void set_device_value(Type* dst, const Type* src)
```

Comment:
I would `const Type& src` in this case - even if it is a bit inconsistent with the async case.

Reply:
done